docs/parse_argparse.py

```diff
@@ -197,88 +203,94 @@ def design_main_file(fpath, rst_dirpath):
     rst_dirpath : string
         path to the directory where script.rst files will be written,
         e.g. '</path/to/docs/source/scripts/directory>'
-
-    Returns
-    -------
-    string
-        path to the `_main.rst` file corresponding to each directory
-        e.g. '</path/to/docs/source/scripts/directory/directory_main.rst>' 
     """
     # extract "/home/siddharth/Autometa/autometa/common/external"
     dir_path = os.path.dirname(os.path.abspath(fpath))
-    rst_main_dir = os.path.basename(dir_path)  # extraxt "external"
-    rst_main_file = rst_main_dir + "_main.rst"  # making a file external_main
-    main_file_fpath = os.path.join(rst_dirpath, rst_main_file)
-    outlines = ""
-    if os.path.exists(main_file_fpath):
-        return main_file_fpath
-    with open(main_file_fpath, "w") as fh:
-        frmt_header = "="*len(rst_main_dir)
-        outlines += f"{frmt_header}\n{rst_main_dir}\n{frmt_header}\n\n.. toctree::\n"
+    main_rst_dir = os.path.basename(dir_path)  # extraxt "external"
+    main_rst_file = main_rst_dir + "_main.rst"  # making a file external_main
+    main_rst_fpath = os.path.join(rst_dirpath, main_rst_file)
+    if not os.path.exists(main_rst_fpath):
+        frmt_header = "="*len(main_rst_dir)
+        outlines = f"{frmt_header}\n{main_rst_dir}\n{frmt_header}\n\n.. toctree::\n"
         outlines += f"\t :maxdepth: 2\n\t :caption: Table of Contents\n\n"
-        path_recorded = ""
-        for __, dirnames, _ in os.walk(dir_path):
-            if path_recorded == main_file_fpath:
-                continue
-            for dirname in dirnames:
-                if "__" in dirname:  # to prevent __pychache__ folders from being considered
-                    continue
-                dir_main = dirname + "_main"  # external_main
-                path_recorded = main_file_fpath
-                lines = f"\n\t {dirname}/{dir_main}"
-                outlines += lines
-        fh.write(outlines)
-    return (main_file_fpath)
+        with open(main_rst_fpath, "w") as fh:
+            fh.write(outlines)
```

Comment:
The reason it is inside the `if` statement is because the program will create `main_rst_fpath` for each `fpath` i.e. for each and every python script. If I place then:
1. It leads to an error as now outlines is referenced before assignment
2. It will open  `main_rst_fpath` as many times as there as `fpaths` which is a bit ineffecient